plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -251,6 +252,9 @@ static String getParametersString(Map<String, String> parametersMap) {
     return resultBuilder.toString();
   }
 
+  // To prevent showing multiple opt-in dialogs. See OptInDialog#open() and OptInDialog#close().
```

Comment:
Ah, I remembered why I did this weird thing to make the OptInDialog manage the flag. We have this assumption that once a dialog is shown then the choice is made even though the dialog is just closed (e.g., a user just escape it). In this case, we don't need to care to restore the flag. However, once we chance our policy and decide to show the dialog again if a user didn't explicitly made a choice, this won't work.

So, I'll let you make the final decide. Are we good to set the flag only once with the name `dialogOpenedStatus`?